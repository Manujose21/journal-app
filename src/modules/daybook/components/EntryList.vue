<template>
  <div class="entry-list-container">
    <div class="px-2 pt-2">
      <input
      type="text"
      class="form-control"
      placeholder="Buscar"
      > 
    </div>
   
    <div class="entry-list mt-2">
    <Entry 
      v-for="item in 10"
      :key="item"
    />
    </div>
  </div>
</template>

<script>
import { defineAsyncComponent } from 'vue';

  export default {
    components:{
      Entry: defineAsyncComponent(() => import("./EntryComponent.vue"))
    }
  }
</script>

<style scoped>
  .entry-list-container{
    /* padding: 1rem; */
    border-right: 1px solid #a3a1a1;
    height: calc( 100vh - 56px );
  }
  .entry-list{
    height: calc( 100vh - 120px );
    overflow-y: auto;
  }
</style>