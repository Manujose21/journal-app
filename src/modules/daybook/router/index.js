
export default {
  name: "daybook",
  component: () => import(/* webpackChunkName: "daybook" */ '@/modules/daybook/layout/DaybookLayout'),
  children: [
    {
      path: "",
      name: "not-entry",
      component: () => import(/* webpackChunkName: "daybook-not-entry" */ '@/modules/daybook/views/NotEntrySelected')

    },
    {
      path: ":entry",
      name: "entry-view",
      component: () => import(/* webpackChunkName: "entry-view" */ '@/modules/daybook/views/EntryView'),
      props: ( route ) =>{
        return {
          id: route.params.entry
        }
      }
      
      
    }
  ]
}