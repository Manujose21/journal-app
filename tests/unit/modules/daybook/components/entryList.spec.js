 
import { shallowMount } from "@vue/test-utils"
import { createStore } from 'vuex'
import { daybookState } from '../../../mock-data/test-data.js'
import daybook from '@/modules/daybook/store/daybookStore'
import EntryList from '@/modules/daybook/components/EntryList.vue'

const createVuexStore = ( initialState ) => 
createStore({
  modules: {
    daybook : {
      ...daybook,
      state: { ...initialState }
    }
  }
})

 describe('' , () =>{  

  const store = createVuexStore( daybookState )
  const mockRouter = {
    push: jest.fn()
  }

  let wrapper ;

  beforeEach(() => {

    jest.clearAllMocks()

    wrapper = shallowMount(EntryList, {
      global:{
        mocks:{
          $router: mockRouter
        },
        plugins: [ store ]
      },
    })

  })

  it('snapshot', () => {
    expect(wrapper.findAll('entry-stub').length).toBe(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('search entry', async () => {

    const input = wrapper.find('input')

    await input.setValue('axios')

    expect(wrapper.findAll('entry-stub').length).toBe(1)

  })

  it('new entry button', () =>{

    wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith({name: 'entry-view', params:{ entry: 'new' } })
    

  })

 })