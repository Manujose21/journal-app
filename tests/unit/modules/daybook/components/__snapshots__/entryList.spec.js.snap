// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` snapshot 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar"></div>
  <div class="d-flex justify-content-center mt-2"><button class="btn btn-primary"> Nueva entrada <i class="fa fa-plus-circle"></i></button></div>
  <div class="entry-list mt-2">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
